pandas/core/internals/managers.py

```diff
@@ -798,24 +812,43 @@ def as_array(self, transpose: bool = False) -> np.ndarray:
             arr = np.empty(self.shape, dtype=float)
             return arr.transpose() if transpose else arr
 
+        should_copy = copy or na_value is not lib.no_default
+
         if self._is_single_block and self.blocks[0].is_datetimetz:
             # TODO(Block.get_values): Make DatetimeTZBlock.get_values
             # always be object dtype. Some callers seem to want the
             # DatetimeArray (previously DTI)
             arr = self.blocks[0].get_values(dtype=object)
+        elif self._is_single_block and self.blocks[0].is_extension:
```

Comment:
you might be able to remove the block aboev (is_datetimetz) as that is an extension block already

Reply:
Indeed, the `DatetimeArray.to_numpy` behaviour seems correct for the default (having object dtype with timestamps instead of datetime64)

Reply:
Nice 👍